libflux/go/libflux/analyze.go

```diff
@@ -20,20 +20,19 @@ type SemanticPkg struct {
 }
 
 // MarshalFB serializes the given semantic package into a flatbuffer.
-func (p *SemanticPkg) MarshalFB() ([]byte, error) {
+func (p *SemanticPkg) MarshalFB() ([]byte, uint, func(), error) {
```

Comment:
I wonder if it would be better to return a pointer to a struct called `Buffer` or something that implements the `Freeable` interface, and use `SetFinalizer` similar to how `ASTPkg` and `SemanticPkg` do to ensure that it's always cleaned up, even if the caller forgets to do it.

`Buffer` could encapsulate the `offset` logic too.

Thoughts?

This would also make the return type of this function a bit simpler

Reply:
Yeah, that will be a good idea.